Gems/GraphCanvas/Code/StaticLib/GraphCanvas/Widgets/NodeSearcher/NodeSearcherTableModel.cpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/std/sort.h>
+
+#include <GraphCanvas/Widgets/NodeSearcher/NodeSearcherTableModel.h>
+#include <GraphCanvas/Components/SceneBus.h>
+#include <GraphCanvas/Components/VisualBus.h>
+
+namespace GraphCanvas
+{
+    ////////////////////////
+    // NodeTableSourceModel
+    ////////////////////////
+
+    NodeTableSourceModel::NodeTableSourceModel()
+    {
+
+    }
+
+    NodeTableSourceModel ::~NodeTableSourceModel()
+    {
+
+    }
+
+    void NodeTableSourceModel::SetActiveScene(const AZ::EntityId& graphId)
+    {
+        GraphCanvas::SceneNotificationBus::Handler::BusDisconnect();
+        GraphCanvas::NodeTitleNotificationsBus::MultiHandler::BusDisconnect();
+        m_activeGraph = graphId;
+        GraphCanvas::SceneNotificationBus::Handler::BusConnect(m_activeGraph);
+        
+
+        {
+            AZStd::lock_guard<AZStd::mutex> lock(m_nodexMtx);
+            m_nodes.clear();
+            m_nodeNames.clear();
+            GraphCanvas::SceneRequestBus::EventResult(m_nodes, m_activeGraph, &GraphCanvas::SceneRequests::GetNodes);
+            for (auto& nodeId: m_nodes)
+            {
+                AZStd::string nodeName;
+                GraphCanvas::NodeTitleRequestBus::EventResult(nodeName, nodeId, &GraphCanvas::NodeTitleRequests::GetTitle);
+                m_nodeNames[nodeId] = nodeName;
+
+                GraphCanvas::NodeTitleNotificationsBus::MultiHandler::BusConnect(nodeId);
+            }
+            ReSortNodes();
+        }
+        
+        layoutChanged();
+    }
+
+    int NodeTableSourceModel::rowCount([[maybe_unused]] const QModelIndex& parent) const
+    {
+        return static_cast<int>(m_nodes.size());
+    }
+
+    int NodeTableSourceModel::columnCount([[maybe_unused]] const QModelIndex& index) const
+    {
+        return CD_Count;
+    }
+
+    QVariant NodeTableSourceModel::data(const QModelIndex& index, int role) const
+    {
+        return index.isValid() && role == Qt::DisplayRole ? m_nodeNames.at(m_nodes[index.row()]).c_str() : QVariant();
+    }
+
+    bool NodeTableSourceModel::setData([[maybe_unused]] const QModelIndex& index, [[maybe_unused]] const QVariant& value, [[maybe_unused]] int role)
+    {
+        return true;
+    }
+
+    QVariant NodeTableSourceModel::headerData([[maybe_unused]] int section, Qt::Orientation orientation, int role) const
+    {
+        return role == Qt::DisplayRole && orientation == Qt::Horizontal ? QString("NodeName") : QVariant();
```

Comment:
Does this need to display header since there is only one column? Should be clear that it is the node name. I also don't see headers in any of the other invisible windows in the screenshots.